kubernetes-client/src/main/java/io/fabric8/kubernetes/client/internal/readiness/Readiness.java

```diff
@@ -59,10 +65,30 @@ public static boolean isReady(HasMetadata item) {
     if (isReadiableKubernetesResource(item)) {
       return isKubernetesResourceReady(item);
     } else {
-      throw new IllegalArgumentException("Item needs to be one of [Node, Deployment, ReplicaSet, StatefulSet, Pod, ReplicationController], but was: [" + (item != null ? item.getKind() : "Unknown (null)") + "]");
+      return handleNonReadinessApplicableResource(item, getReadinessApplicableResourcesList());
     }
   }
 
+  public static boolean handleNonReadinessApplicableResource(HasMetadata item, List<String> readinessApplicableResources) {
+    boolean doesItemExist = Objects.nonNull(item);
+    String readinessApplicableResourcesStr = String.join(",", readinessApplicableResources);
+    logger.warn("{} is not a Readiableresource. It needs to be one of [{}]",
+      doesItemExist ? item.getKind() : "Unknown", readinessApplicableResourcesStr);
+    return doesItemExist;
+  }
+
+  public static List<String> getReadinessApplicableResourcesList() {
+    List<String> readinessApplicableResources = new ArrayList<>();
+    readinessApplicableResources.add("Node");
+    readinessApplicableResources.add("Deployment");
+    readinessApplicableResources.add("ReplicaSet");
+    readinessApplicableResources.add("StatefulSet");
+    readinessApplicableResources.add("Pod");
+    readinessApplicableResources.add("ReplicationController");
+
+    return readinessApplicableResources;
+  }
```

Comment:
If I'm not mistaken, you're basically constructing **the same** String for each method invocation.

Just create a static class variable `String READINESS_APPLICABLE_RESOURCES`

Reply:
In understand that you're doing this because you extend it on the OpenShiftReadiness class, but approach should be the same.

Create a static String in each class, and then implement a **non-static** `String getReadinessApplicableResources()` that will return each of the applicable static strings. Obviously the OpenShiftReadiness method has an `@Override` annotation.